src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,81 @@
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useAppDispatch } from '../../app/hook';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+interface Props {
+  todo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const dispatch = useAppDispatch();
+
+  const clear = () => {
+    dispatch(currentTodoActions.clearTodo());
+  };
+
+  const [user, setUser] = useState<User>();
+  const [loader, setLoader] = useState(false);
+
+  useEffect(() => {
+    setLoader(true);
+
+    getUser(todo.userId)
+      .then(setUser)
+      .finally(() => setLoader(false));
+  }, [todo.userId]);
+
+  function handleClose() {
+    clear();
+  }
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {loader ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{todo.id}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleClose}
+            />
+          </header>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {todo.title}
+            </p>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            <p className="block" data-cy="modal-user">
+              {todo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+              {' by '}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
Ensure that the `user` object is always defined before accessing its properties like `email` and `name`. You might want to add a check or a fallback to handle cases where `user` is undefined.